activerecord/lib/active_record/autosave_association.rb

```diff
@@ -106,7 +108,7 @@ module ActiveRecord
   #   post.comments[0].body = 'hi everyone'
```

Comment:
I think the way you wrote this in your example, with a local variable, is clearer than using `[0]`. Could you update this one to use the same style?

I think having the two examples as similar as possible will make it clearer what the difference in behaviour is.